src/snowflake/cli/_plugins/nativeapp/version/version_processor.py

```diff
@@ -156,11 +159,7 @@ def add_new_patch_to_version(self, version: str, patch: Optional[int] = None):
                 add_version_query, cursor_class=DictCursor
             )
 
-            show_row = find_first_row(
-                result_cursor,
-                lambda row: row[VERSION_COL] == unquote_identifier(version),
-            )
```

Comment:
we don't actually need this, the query only returns one row

Reply:
We do need it. Because of how the query is made (with like), we can return multiple rows that have the "same" identifier after the like rules are applied. So we need to filter back down to the one we're actually dealing with.

I think `"Name"` and `name` is one such example.

Reply:
the query here is
```python
                    alter application package {self.package_name}
                        add patch {patch if patch else ""} for version {version}
                        using @{self.stage_fqn}
```
I don't see how that issue could happen here

Reply:
Oh sorry, drive-by review. Let me look into it a bit more closely.

Reply:
Yep, confirmed, even when there are multiple verisons only the one in question is returned.